### PR TITLE
karmada operator: add karmada operator readme

### PR DESCRIPTION
**Pull request by calvin0327**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**
Add karmada operator README to introduce how to use karmada operator.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by calvin0327** [comment]
@lonelyCZ @Poor12 /cc

File: operator/README.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# Karmada operator
+
+## Overview
+
+The Karmada operator is a project of the Karmada community designed to manage the lifecycle of Karmada. we can operate CR to install, upgrade and delete the Karmada instance. It is a centrally managed controller that can install karmada instance not only in the local cluster, but also in a remote cluster. It is convenient to manage multiple karmada instances in a global cluster.
```
Comment (Poor12) [inline operator/README.md:5]: We can

Comment (Poor12) [inline operator/README.md:5]: We need to unify the `Karmada instance` and `karmada instance`.

File: operator/README.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+```shell
+kubectl get po -n karmada-system
+NAME                               READY   STATUS    RESTARTS   AGE
+karmada-operator-5b7f485c5-g5lj5   1/1     Running   0          26s
+```
+
+## Create Karmada
+
+`Karmada` CR represents a karmada instance, which is a Namespace-level resource. <br>
```
Comment (Poor12) [inline operator/README.md:34]: We need to unify `Karmada` CR or Karmada CR or karmada CR.

File: operator/README.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+kubectl delete karmada karmada-demo -n test
+```
+
+## Custom Karmada CR
+
+Flexibility to configure CR to install Karmada instances. For details, please See [Karmada.yaml](./config/samples/karmada.yaml) <br>
+
+### set karmada component replica
+
+The replica of all karmada components is modifiable. scale etcd pod replica to 3, e.g:
```
Comment (windsonsea) [inline operator/README.md:102]:
```suggestion
## Custom Karmada CR

This feature allows you to configure the Karmada CR to install Karmada instances flexibly. For details, see [karmada.yaml](./config/samples/karmada.yaml).

### Set Karmada component replicas

The replicas of all Karmada components can be modified. For example, to scale the etcd pod replicas to 3:
```

File: operator/README.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+spec:
+  components:
+    etcd:
+      local:
+        replicas: 3
+```
+
+### karmada-apiserver service type
+
+The service type of karmada-apiserver is `ClusterIP` by default, update it to `NodePort`, e.g:
```
Comment (windsonsea) [inline operator/README.md:119]:
```suggestion
The service type of karmada-apiserver is `ClusterIP` by default. You can change it to `NodePort`:
```

File: operator/README.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  name: karmada-demo
+  namespace: test
+spec:
+  components:
+    etcd:
+      local:
+        replicas: 3
+```
+
+### karmada-apiserver service type
```
Comment (windsonsea) [inline operator/README.md:117]:
```suggestion
### Change karmada-apiserver service type
```

File: operator/README.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+karmadaAPIServer:
+  imageRepository: registry.k8s.io/kube-apiserver
+  imageTag: v1.25.4
+  replicas: 1
+  serviceType: NodePort
+  serviceSubnet: 10.96.0.0/12
+...
+```
+
+### karmada-apiserver SANs
```
Comment (windsonsea) [inline operator/README.md:132]:
```suggestion
### Add karmada-apiserver SANs
```

File: operator/README.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+Add extra SANs to karmada-apiserver certificate. e.g:
```
Comment (windsonsea) [inline operator/README.md:134]:
```suggestion
You can add more SANs to karmada-apiserver certificate:
```

**Review by windsonsea — COMMENTED** [review]
Have a try to improve text :)

**Comment by calvin0327** [comment]
> Have a try to improve text :)

thanks for your review.

File: operator/README.md (hunk 1 above)
Comment (calvin0327) [reply operator/README.md]: I'm with you.

**Comment by windsonsea** [comment]
LGTM. No more idea. cc @Poor12

File: operator/README.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+A Karmada instance can be deleted once the Karmada CR is deleted.
+
+```shell
+kubectl delete karmada karmada-demo -n test
+```
+
+## Custom Karmada CR
+
+This feature allows you to configure the Karmada CR to install Karmada instances flexibly. For details, see [karmada.yaml](./config/samples/karmada.yaml).
```
Comment (Poor12) [inline operator/README.md:96]: Could we add some more use cases?
eg. install karmada-descheduler/custom labels.

**Comment by calvin0327** [comment]
@windsonsea @Poor12 
I added some content to the doc:

* add example to install descheduler
* how to contribute
* document optimization

Any other comments?

**Comment by windsonsea** [comment]
Ok, I have revised it again. cc @Poor12 to review.

**Comment by Poor12** [comment]
Thanks all.
/lgtm
/approve

**Pull request merged**